openslides_backend/action/actions/motion/update.py

```diff
@@ -108,17 +108,18 @@ def set_recommendation_extension_reference_ids(
         possible_rerids = RECOMMENDATION_EXTENSION_REFERENCE_IDS_PATTERN.findall(
             instance["recommendation_extension"]
         )
+        motion_ids = []
         for fqid_str in possible_rerids:
             collection, id_ = fqid_str.split(KEYSEPARATOR)
             if collection != "motion":
                 raise ActionException(f"Found {fqid_str} but only motion is allowed.")
-            exists = self.datastore.exists(
-                collection=Collection(collection),
-                filter=FilterOperator("id", "=", int(id_)),
-            )
-            if exists:
+            motion_ids.append(int(id_))
+        gm_request = GetManyRequest(Collection("motion"), motion_ids, ["id"])
+        gm_result = self.datastore.get_many([gm_request]).get(Collection("motion"), {})
+        for motion_id in motion_ids:
+            if motion_id in gm_result:
```

Comment:
Iterate over the `gm_result` directly to save time

Reply:
Done.